### PR TITLE
An aside on function composition

### PR DESCRIPTION
**Pull request by sneakers-the-rat**
Continuing: https://github.com/pyOpenSci/lessons/issues/58

Just a quick aside with some links to some wikipedia pages for further reading. I think this is good to introduce here for reasons that are along the same lines as what i was saying re: "try/except? that's great! so now i'm going to wrap everything in try/except and now no more errors!" - we don't want the students to overgeneralize and pack everything into big functions :)

### CODE REVIEW

File: code-workflow-logic/python-functions-multi-parameters.md [hunk 1]
```diff
@@ -189,6 +189,16 @@ def mean_mm_to_in(data_mm, axis_value):
 
 Note that the function could be written to convert the values first and then calculate the mean. However, given that the function will complete both tasks and return the mean values in the desired units, it is more efficient to calculate the mean values first and then convert just those values, rather than converting all of the values in the input array. 
 
+````{tip}
+Typically functions should have a [single purpose](https://en.wikipedia.org/wiki/Separation_of_concerns), and be [composed](https://en.wikipedia.org/wiki/Function_composition_(computer_science)) to implement higher-order operations. The above function would normally be written without wrapping {func}`numpy.mean` like this:
```
Comment (lwasser) [inline code-workflow-logic/python-functions-multi-parameters.md:193]: OH YES!! functions should do one thing well. i love this!

**Review by lwasser — APPROVED** [review]

**Comment by lwasser** [comment]
thank you!! i'll try my best to slow down a bit as well!!

**Comment by sneakers-the-rat** [comment]
I would never tell you to slow down!

**Comment by lwasser** [comment]
@sneakers-the-rat you are the best. I appreciate you so much!

**Pull request merged**